cli/command/service/logs.go

```diff
@@ -93,6 +93,14 @@ func runLogs(dockerCli *command.DockerCli, opts *logsOptions) error {
 			return err
 		}
 		task, _, err := cli.TaskInspectWithRaw(ctx, opts.target)
+		if err != nil {
+			if client.IsErrTaskNotFound(err) {
+				// if the task isn't found, rewrite the error to be clear
+				// that we looked for services AND tasks and found none
+				err = fmt.Errorf("no such task or service")
```

Comment:
It doesn't *really* work to specify whether we're looking for a service or a task, because we first check for service and then we check for task, and there isn't really a way to differentiate the two based on the user's input.